app/components/users_modal/users_modal.tsx

```diff
@@ -0,0 +1,234 @@
+
+    const isSearch = Boolean(term);
+
+    const loadedProfiles = useCallback(({users}: {users?: UserProfile[]}) => {
+        if (mounted.current) {
+            if (users && !users.length) {
+                next.current = false;
+            }
+
+            page.current += 1;
+            setLoading(false);
+            dispatchProfiles({type: 'add', values: users});
+        }
+    }, []);
+
+    const handleButtonTap = useCallback(async (selectedId?: {[id: string]: boolean}) => {
+        if (startingButtonAction) {
+            return;
+        }
+
+        setStartingButtonAction(true);
+
+        const idsToUse = selectedId ? Object.keys(selectedId) : Object.keys(selectedIds);
+        const success = idsToUse.length === 0 ? false : await onButtonTap();
+
+        if (success) {
+            close();
+            return;
+        }
+
+        setStartingButtonAction(false);
+    }, [onButtonTap, selectedIds, startingButtonAction]);
+
+    const handleGetProfiles = useCallback(debounce(() => {
+        if (next.current && !loading && !term && mounted.current) {
+            setLoading(true);
+            getProfiles().then(loadedProfiles);
+        }
+    }, 100), [getProfiles, loading, term]);
+
+    const data = useMemo(() => {
+        if (isSearch) {
+            const exactMatches: UserProfile[] = [];
+            const filterByTerm = (p: UserProfile) => {
+                if (selectedCount > 0 && p.id === currentUserId) {
+                    return false;
+                }
+
+                if (p.username === term || p.username.startsWith(term)) {
+                    exactMatches.push(p);
+                    return false;
+                }
+
+                return true;
+            };
+
+            const results = filterProfilesMatchingTerm(searchResults, term).filter(filterByTerm);
+            return [...exactMatches, ...results];
+        }
+        return profiles;
+    }, [isSearch && selectedCount, isSearch && searchResults, profiles, term]);
+
+    const updateNavigationButtons = useCallback(async (startEnabled: boolean) => {
+        const closeIcon = await CompassIcon.getImageSource('close', 24, theme.sidebarHeaderTextColor);
+        setButtons(componentId, {
+            leftButtons: [{
+                id: CLOSE_BUTTON,
+                icon: closeIcon,
+                testID: 'close.button',
+            }],
+            rightButtons: [{
+                color: theme.sidebarHeaderTextColor,
+                id: ACTION_BUTTON,
+                text: formatMessage(buttonText),
+                showAsAction: 'always',
+                enabled: startEnabled,
+                testID: 'action.button',
+            }],
+        });
+    }, [buttonText, locale, theme]);
+
+    useEffect(() => {
+        mounted.current = true;
+        updateNavigationButtons(false);
+        handleGetProfiles();
+        return () => {
+            mounted.current = false;
+        };
+    }, []);
+
+    useEffect(() => {
```

Comment:
This effect runs on the first render (as any effect) and set the navigation buttons, but the previous effect also set the navigation buttons.

Either this effect should be a `useDidUpdate`, or we can remove the `updateNavigationButtons` call on the previous effect. Up to you.